3rdparty/libconfig/scanner.c

```diff
@@ -589,43 +843,84 @@ static const flex_int32_t yy_rule_can_match_eol[46] =
 
 #define YY_NO_INPUT // Suppress generation of useless input() function
 
-static unsigned long long fromhex(const char *s)
+static unsigned long long fromhex(const char *p)
 {
-#ifdef __MINGW32__
+  unsigned long long val = 0;
+
+  if (*p != '0' || (p[1] != 'x' && p[1] != 'X')) {
+    return 0;
+  }
 
-  /* MinGW's strtoull() seems to be broken; it only returns the lower
-   * 32 bits...
-   */
+  for (p += 2; isxdigit(*p) || *p == '_'; ++p) {
+    if (*p != '_') {
+      val <<= 4;
+      val |= ((*p < 'A') ? (*p & 0xF) : (9 + (*p & 0x7)));
+    }
+  }
+
+  return val;
+}
 
-  const char *p = s;
+static unsigned long long frombin (const char *p)
+{
   unsigned long long val = 0;
 
-  if(*p != '0')
-    return(0);
+  if (*p != '0' || (p[1] != 'b' && p[1] != 'B')) {
+    return 0;
+  }
+
+  for (p += 2; *p == '0' || *p == '1' || *p == '_'; ++p) {
+    if (*p != '_') {
+      val <<= 1;
+      val |= *p == '0' ? 0 : 1;
```

Comment:
```suggestion
      val |= ((*p == '0') ? 0 : 1);
```

Reply:
scanner.c and grammar.c are both generated files... but you're right; I forgot to re-generate them after applied the changed you requested